src/app/caesar-portal/group-list/group-list.component.ts

```diff
@@ -16,55 +20,96 @@ import { BsModalRef } from 'ngx-bootstrap/modal/modal-options.class';
 
 export class GroupListComponent implements OnInit {
   groups: Group[];
+  filteredGroups: Group[];
   bsModalRef: BsModalRef;
   firstItem = 0;
   lastItem = 9;
   itemsPerPage = 10;
   currentPage = 1;
-  groupsQuantity: number;
   myGroupsFilter: boolean = false;
   GroupProgressStatus: typeof GroupStatus = GroupStatus;
   groupStatus: GroupStatus = GroupStatus.Current;
+  locations: string[] = [];
+  subscription: Subscription;
 
-  constructor(private groupService: GroupService, private modalService: BsModalService) { }
+  constructor(
+    private groupService: GroupService,
+    private modalService: BsModalService,
+    private activateRoute: ActivatedRoute,
+    private locationService: LocationService
+  ) { }
 
   ngOnInit() {
-    this.showAllGroups();
+    this.subscription = this.activateRoute.params.subscribe(params => {
+      if (params['location']) {
+        this.locations = params['location'].split('+');
+      }
+      this.locations.length > 0
+        ? this.getGroupsByLocation(this.locations)
+        : this.getCurrentLocationGroups();
+    });
   }
 
-  private showAllGroups() {
-    this.groupService.getAll().subscribe(
+ 
+
+  private getCurrentLocationGroups() {
```

Comment:
why all methods is private? if your view uses it it should be public

Reply:
View is using openDeleteDialog, changeProgressStatus,onPageChange and showMyGroups, make them public and everything else private ?